sympy/utilities/iterables.py

```diff
@@ -523,9 +521,15 @@ def ibin(n, bits=0, str=False):
     ['000', '001', '010', '011', '100', '101', '110', '111']
 
     """
+    if n < 0:
+        raise ValueError("negetive numbers are not allowed")
+
+    if n.bitlength() is not None and n.bitlength() > bits:
```

Comment:
`bit_length`: https://docs.python.org/3.7/library/stdtypes.html#int.bit_length

Reply:
Done. Thank you.